pilot/pkg/xds/debug.go

```diff
@@ -687,11 +688,20 @@ func (s *DiscoveryServer) ConfigDump(w http.ResponseWriter, req *http.Request) {
 		return
 	}
 	if ts := s.getResourceTypes(req); len(ts) != 0 {
+		if con.proxy.IsZTunnel() {
+			istiolog.Warnf("proxy %v is not envoy based", proxyID)
+		}
 		dump := s.getConfigDumpByResourceType(con, ts)
 		writeJSON(w, dump, req)
 		return
 	}
 
+	if con.proxy.IsZTunnel() {
+		workloads := s.debugWorkloads(con.proxy.LastPushContext.AmbientIndex)
```

Comment:
I think we want to return the result of the workload generator (using getConfigDumpByResourceType L694). This part is already exposed  under ambientz and is mostly useless for ztunnel (only used for waypoint)

Reply:
any thoughts one way or the other about returning the entire workload vs the simplified struct like in ambientz?  The ztunnel config_dump API is already returning a different internal struct, so the outputs cannot be diffed directly to see what's missing

Reply:
/config_dump should take in a list of types and a proxy and return the XDS for that type+proxy combination.

Reply:
yeah, ideally, formatted similarly to what the ztunnel receives so we can easily compare the configs (from istiod vs ztunnel).